x-pack/plugins/security/server/routes/api/public/roles/get.js

```diff
@@ -11,11 +11,31 @@ import { wrapError } from '../../../../lib/errors';
 export function initGetRolesApi(server, callWithRequest, routePreCheckLicenseFn, application) {
 
   const transformKibanaApplicationsFromEs = (roleApplications) => {
-    return roleApplications
-      .filter(roleApplication => roleApplication.application === application)
-      .filter(roleApplication => roleApplication.resources.length > 0)
-      .filter(roleApplication => roleApplication.resources.every(resource => resource === ALL_RESOURCE))
-      .map(roleApplication => ({ privileges: roleApplication.privileges }));
+    const roleKibanaApplications = roleApplications
+      .filter(roleApplication => roleApplication.application === application);
+
+    const resourcePrivileges = _.flatten(roleKibanaApplications
+      .map(({ resources, privileges }) => resources.map(resource => ({ resource, privileges })))
+    );
+
+    return resourcePrivileges.reduce((result, { resource, privileges }) => {
+      if (resource === ALL_RESOURCE) {
+        result.global = _.uniq([...result.global, ...privileges]);
+        return result;
+      }
+
+      const spacePrefix = 'space:';
```

Comment:
This is hard-coded here, and I wanted to address it's ugliness in the security spaces PR. We'll need a way to assign the `space_all` privilege at a space, so I think they're gonna be related and this work is already started in the other branch.